datadog/dogstatsd/base.py

```diff
@@ -423,26 +441,51 @@ def __init__(
 
         self._reset_buffer()
 
-        # This lock is used for all cases where client configuration is being changed: buffering, sender mode.
+        # This lock is used for all cases where client configuration is being changed: buffering,
+        # aggregation, sender mode.
         self._config_lock = RLock()
 
-        # If buffering is disabled, we bypass the buffer function.
-        self._send = self._send_to_buffer
         self._disable_buffering = disable_buffering
-        if self._disable_buffering:
-            self._send = self._send_to_server
-            log.debug("Statsd buffering is disabled")
+        self._disable_aggregating = disable_aggregating
 
+        self._flush_interval = flush_interval
+        self._aggregation_flush_interval = aggregation_flush_interval
+        # We make the _buffering_flush_thread and _aggregation_flush_thread a list so that it is a mutable object,
```

Comment:
is there a reason we cannot continue using `self._flush_thread` in favor of the list thread containers? 
Since aggregation and buffering won't be enabled at the same time (at this point) `self._flush_thread` will only be occupied by one or the other routine.

Reply:
No I think that works as well, I just thought designing it like this would be more flexible and at the time I wrote the code I dont think I knew whether or not they would be allowed at the same time.

Reply:
Makes sense. In that case - let's do the simple thing for now and reuse `self. _flush_thread`.

Reply:
Sounds good